### PR TITLE
Default to closing ES indices

### PR DESCRIPTION
**Pull request by emord**
This changes `prune_elastic_indices` to flushing and closing an index instead of deleting. Closing an index provides all the same benefits of deleting except that it doesn't delete it from the disk.

Will also have a followup PR to make sure this doesn't delete UCR indices.

@czue

### CODE REVIEW

**Comment by czue** [comment]
> Will also have a followup PR to make sure this doesn't delete UCR indices.

yikes, good call.

**Comment by czue** [comment]
looks good but one question - after you close the indexes do they still show up anywhere? I think one of the main benefits of pruning is freeing disk space back up so we should still have a workflow that supports that

**Comment by emord** [comment]
they still show up in eshead and its very easy to recover them by opening them. 

I moved deleting to an option https://github.com/dimagi/commcare-hq/pull/12992/commits/6742313e2b56396408346ebca6021475355ac8e6#diff-64ef4ec25c6dc1da08b99572c1a8625fR16

I decided to do pruning by default after noticing htis issue: https://github.com/dimagi/commcare-hq/pull/12992/commits/6742313e2b56396408346ebca6021475355ac8e6#diff-64ef4ec25c6dc1da08b99572c1a8625fR24

basically if a reindex is in progress and this is run either the old or new index will be deleted (depending on what code branch you run the management command in)

**Comment by czue** [comment]
yeah all makes sense. I think what I meant to ask more explicitly was whether this command does already or could in the future be modified to delete closed indices. like does this call: `es.indices.get_aliases().keys()` include closed already or could we update it so that it does? 

based on the fact that they show up in HEAD I assume there must be a way so this seems ok for now.

**Comment by emord** [comment]
Oh yeah. this will delete closed indicies. I might add a command to only delete closed indices in the future
